test/tests/layer.js

```diff
@@ -82,7 +89,30 @@ suite("d3.layer", function() {
 			this.layer.draw([]);
 			assert(this.insert.calledOn(this.dataBind.returnValues[0].enter.returnValues[0]));
 		});
-		
+		test("by default removes exiting nodes from the DOM", function() {
+			this.layer.draw([1]);
+			assert.equal(this.layer.selectAll("g").size(), 1);
+			this.layer.draw([]);
+			assert.equal(this.layer.selectAll("g").size(), 0);
+		});
+		test("invokes the provided `remove` method in the context of the layer's bound 'exit' selection", function() {
+			var updating, exiting;
+
+			this.layerWithRemove.draw([1, 2, 3]);
+			this.layerWithRemove.draw([]);
+
+			updating = this.dataBind.returnValues[1];
+			exiting = updating.exit.returnValues[0];
+
+			assert(this.remove.calledOn(exiting));
```

Comment:
I think this is definitely indicative of a naming issue. As proposed, "remove" describes functionality for each data-bound element individually when its associated data point is removed from the set. I can see how its generic name and association with the Layer instance itself suggests that the Layer instance is the target.

Unfortunately, a consistent solution will involve a breaking API change. This is because "insert" *already* suffers from the problem you are identifying (in the same way, it is not the insertion logic for the Layer instance, but the insertion logic for each new data-bound element).

What we would need is some clear, concise nomenclature for "data-bound element". I'm going to use the word "point" for demonstration purposes (even though I admit that is a terrible choice).

The two methods might be better named `insertPoint` and `removePoint` to more clearly describe their targets.

```js
d3.select('svg').append('g').layer({
  dataBind: myDataBindFunction,
  insertPoint: myFunctionForInsertingASingleNewDataBoundElement,
  removePoint: myFunctionForRemovingASingleDataBoundElement
});
```

If we found a better name than "point", do you think this would clear things up?